### ISSUE TITLE
Remove BearerTokenRequestMatcher

### ISSUE THREAD

**Issue by jgrandja** [jzheaux/spring-security#19]
I don't think `BearerTokenRequestMatcher` is needed at this point.

It's only referenced in `OAuth2ResourceServerConfigurer` and set in `ExceptionHandlingConfigurer` and `CsrfConfigurer`.

We can just pass a lambda reference of `BearerTokenResolver.resolve` instead.

Also, the `BearerTokenAuthenticationFilter` does use `BearerTokenRequestMatcher`. It directly uses `BearerTokenResolver`.

**Comment by jzheaux** [comment]
No problem removing. Less stuff for the PR.

Note, the reason that the filter doesn't use the request matcher is because it isn't matching the request--it actually needs the token to begin the authentication procedure.

**Comment by jgrandja** [comment]
> Note, the reason that the filter doesn't use the request matcher is because it isn't matching the request--it actually needs the token to begin the authentication procedure.

Yeah, makes sense. The Filter doesn't need the `RequestMatcher` in this instance
